### PR TITLE
move some optimizers to level1

### PR DESCRIPTION
**Pull request by askhade**
**Description**: Move 3 transformers to level 1 where they really belong

**Motivation and Context**
- Level1 Optimizers are the semantics preserving optimizations like node eliminations, node fusions (which fuse nodes into existing onnx nodes). Even after applying level1 transformer we still have a valid onnx graph which can be executed by any onnx runtime as well as any execution provider of ort. (Where as level2 and l3 optimizations are execution provider specific transformations which alter onnx graph in more complex ways than l1 optimizers)

- Initially we made an exception for ConvAddFusion, ConvMulFusion and ConvBNFusion and added them as level2 optimizers because they are not compatible with training. But it turns out training scenario is more complicated than we had originally anticipated and will need more work to handle it properly. In light of this moving these transformers to their correct level.

- Another reason for making this change now is quantization. We want to enable only the fusions which are not EP specific. This is because for quantization scenario we want both the quantization script and the model to be Execution provider agnostic.

### CODE REVIEW

File: onnxruntime/core/optimizer/graph_transformer_utils.cc [hunk 1]
```diff
@@ -99,8 +99,9 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
     case TransformerLevel::Level1: {
       std::unordered_set<std::string> l1_execution_providers = {};
 
-      transformers.emplace_back(std::make_unique<ConstantFolding>(l1_execution_providers));
+      transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(l1_execution_providers));
       transformers.emplace_back(std::make_unique<MatMulAddFusion>(l1_execution_providers));
```
Comment (snnn) [inline onnxruntime/core/optimizer/graph_transformer_utils.cc]: transformers.emplace_back(std::make_unique<MatMulAddFusion>(l1_execution_providers));

File: onnxruntime/core/optimizer/graph_transformer_utils.cc [hunk 2]
```diff
@@ -35,18 +35,19 @@ std::vector<std::unique_ptr<RewriteRule>> GenerateRewriteRules(TransformerLevel
   std::vector<std::unique_ptr<RewriteRule>> rules;
   switch (level) {
     case TransformerLevel::Level1:
-      rules.push_back(onnxruntime::make_unique<EliminateIdentity>());
-      rules.push_back(onnxruntime::make_unique<EliminateSlice>());
-      rules.push_back(onnxruntime::make_unique<UnsqueezeElimination>());
-      rules.push_back(onnxruntime::make_unique<EliminateDropout>());
-      rules.push_back(onnxruntime::make_unique<FuseReluClip>());
-      rules.push_back(onnxruntime::make_unique<ShapeToInitializer>());
+      rules.push_back(std::make_unique<EliminateIdentity>());
```
Comment (pranavsharma) [inline onnxruntime/core/optimizer/graph_transformer_utils.cc]: onnxruntime::make_unique

Comment (askhade) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc]: resolved in all files

File: onnxruntime/core/optimizer/graph_transformer_utils.cc (hunk 1 above)
Comment (askhade) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc]: done for all occurrences

**Pull request merged**